Client/src/services/saveLastCityToLS.js

```diff
@@ -0,0 +1,26 @@
+const saveLastCityToLS = (city, id) => {
+  let stuffInLs = JSON.parse(localStorage.getItem('lastCity'));
+
+  function checkIfCityExists() {
+    let exists = Boolean;
+    stuffInLs
+      ? stuffInLs.forEach(lsCity => {
+          if (lsCity.city === city) {
+            return (exists = true);
+          } else return (exists = false);
+        })
+      : (exists = false);
+    return exists;
+  }
+
+  if (checkIfCityExists()) {
+    return;
+  } else if (stuffInLs) {
+    stuffInLs.push({ city: city, cityId: id });
+    localStorage.setItem('lastCity', JSON.stringify(stuffInLs));
+  } else {
+    localStorage.setItem('lastCity', JSON.stringify([{ city: city, cityId: id }]));
+  }
+};
+
+export default saveLastCityToLS;
```

Comment:
Didn't test it but here is the whole code:
```
const saveCityToLocalStorage = (city, id) => {
    try {
        const savedCities = JSON.parse(localStorage.getItem('recentCities')) || [];
        const alreadyStored = savedCities.some(savedCity => savedCity.city === city);
        if(alreadyStored) return;
        savedCities.push({ city: city, cityId: id });
        localStorage.setItem('recentCities', JSON.stringify(savedCities));
    } catch (err) {
        console.error(err);
    }
}
```

Reply:
great!